### PR TITLE
refactor veventservice

### PR DESCRIPTION
**Pull request by georgehrke**
- [x] needs tests

### CODE REVIEW

File: tests/js/unit/services/veventServiceSpec.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+		deferred.resolve({
+			status: 403
+		});
+
+		let called = false;
+		deleteRequest.then(function(result) {
+			fail('Promise was not supposed to succeed')
+		});
+		deleteRequest.catch(function(result) {
+			expect(result).toEqual(4035);
```
Comment (georgehrke) [inline tests/js/unit/services/veventServiceSpec.js]: @tcitworld It doesn't hit the catch block. Do you have any idea why?

Comment (georgehrke) [inline tests/js/unit/services/veventServiceSpec.js]: ✅ solved

File: tests/js/unit/services/veventServiceSpec.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+		}, '');
+
+		deferred.resolve({
+			status: 403
+		});
+		$rootScope.$apply();
+
+		let called = false;
+		deleteRequest.then(function() {
+			fail('Promise was not supposed to succeed')
```
Comment (tcitworld) [inline tests/js/unit/services/veventServiceSpec.js]: Dammit semi-column !

Comment (georgehrke) [inline tests/js/unit/services/veventServiceSpec.js]: 🙈

**Comment by georgehrke** [comment]
please review again @tcitworld :)

**Comment by raghunayyar** [comment]
More coverage, yayy! :+1:

**Pull request merged**